framework/src/variables/MooseVariableFE.C

```diff
@@ -47,6 +47,19 @@ MooseVariableFE<OutputType>::MooseVariableFE(const InputParameters & parameters)
       this->_assembly.lowerDElem());
 }
 
+template <typename OutputType>
+std::string
+MooseVariableFE<OutputType>::componentName(const unsigned int comp) const
+{
+  if (comp >= _count)
+    mooseError("Component index must be less than the number of components of variable ",
+               _var_name);
+  if (std::is_same<OutputType, RealEigenVector>::value)
+    return _subproblem.arrayVariableComponent(_var_name, comp);
```

Comment:
```suggestion
    return this->_subproblem.arrayVariableComponent(_var_name, comp);
```

Try this

Reply:
oh! Let me try.

Reply:
I think that older compilers may not be smart enough to try to grab a member variable in a template like that.